File: Go/water_bottles/1518_test.go
package waterbottles

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_numWaterBottles(t *testing.T) {
	tests := []struct {
		numBottles  int
		numExchange int
		want        int
	}{
		{numBottles: 9, numExchange: 3, want: 13},
		{numBottles: 15, numExchange: 4, want: 19},
	}

	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i), func(t *testing.T) {
			got := numWaterBottles(tt.numBottles, tt.numExchange)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: TypeScript/water_bottles/1518.test.ts
import numWaterBottles from './1518';
import { describe, test, expect } from 'bun:test';

interface tt {
  numBottles: number;
  numExchange: number;
  want: number;
}

const tests: tt[] = [
  { numBottles: 9, numExchange: 3, want: 13 },
  { numBottles: 15, numExchange: 4, want: 19 },
];

describe('water bottles', () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = numWaterBottles(t.numBottles, t.numExchange);
      expect(result).toBe(t.want);
    });
  });
});
